operator/cluster/failoverlogic.go

```diff
@@ -108,13 +108,9 @@ func Failover(identifier string, clientset *kubernetes.Clientset, client *rest.R
 		log.Errorf("could not find pgcluster %s with labels", clusterName)
 		return err
 	}
-	cluster.Spec.UserLabels[config.LABEL_CURRENT_PRIMARY] = targetDepName
-	err = util.PatchClusterCRD(client, cluster.Spec.UserLabels, &cluster, namespace)
-	if err != nil {
-		log.Errorf("failoverlogic: could not patch pgcluster %s with labels", clusterName)
-		return err
-	}
 
+	//update the CRD with the new current primary
+	err = util.CurrentPrimaryUpdate(client, &cluster, target, namespace)
 	return err
```

Comment:
```go
// update the CRD with the new current primary. If there is an error, log it
// here, otherwise return
if err := util.CurrentPrimaryUpdate(client, &cluster, target, namespace); err != nil {
  log.Error(err)
  return err
}

return nil
```

Reply:
updated, thanks!